### PR TITLE
FLA-328: X-Transaction-Id not X-Auth-UserId

### PR DESCRIPTION
**Pull request by sdevalapurkar-bcgov**
# Description

This PR includes the following proposed change(s):

- JIRA: https://justice.gov.bc.ca/jira/browse/FLA-328
- make use of a `transactionId` instead of an `X-Auth-UserId` in requests, make bearer token mandatory in all backend calls
- change user validation to happen when user logs in with frontend
- udpate storage to use `transactionId` and `submissionId` as a key pair.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- local
- jest
- storybook
- backend integration with frontend (flow)

### CODE REVIEW

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java [hunk 1]
```diff
@@ -195,32 +199,26 @@ public SubmissionApiDelegateImpl(
...
-            // TODO: remove this
-            userDetails.setFirstName("firstName");
-            userDetails.setLastName("lastName");
-            userDetails.setMiddleName("middleName");
-            userDetails.setEmail("email");
+            userDetails.setUniversalId(accountDetails.getUniversalId());
+            userDetails.setFirstName(accountDetails.getFirstName());
+            userDetails.setLastName(accountDetails.getLastName());
+            userDetails.setMiddleName(accountDetails.getMiddleName());
+            userDetails.setEmail(accountDetails.getEmail());
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:221]: what is this for? I dont think the userdetails object ever has to have any of these fields populated :)

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/GetSubmissionTest.java [hunk 2]
```diff
@@ -116,7 +109,16 @@ public void setUp() {
...
-        sut = new SubmissionApiDelegateImpl(submissionServiceMock, generateUrlResponseMapperMock, navigationProperties, submissionStoreMock, documentStoreMock);
+        Mockito.when(accountServiceMock.getCsoAccountDetails(Mockito.any()))
+                .thenReturn(AccountDetails
+                        .builder()
+                        .universalId(TestHelpers.CASE_2)
+                        .email(EMAIL + TestHelpers.CASE_2)
+                        .firstName(FIRST_NAME + TestHelpers.CASE_2)
+                        .accountId(BigDecimal.ONE)
+                        .create());
+
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/GetSubmissionTest.java:120]: same question here, about why we need this data being part of `userDetails`

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java (hunk 1 above)
Comment (alexjoybc) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:219]: that's good comment but it is legacy and I will remove that in an other PR

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java [hunk 3]
```diff
@@ -229,8 +225,9 @@ private UserDetails buildUserDetails(Submission submission) {
     }
 
     @Override
-    public ResponseEntity<FilingPackage> getSubmissionFilingPackage(UUID xAuthUserId, UUID id) {
-        Optional<Submission> fromCacheSubmission = this.submissionStore.get(id, xAuthUserId);
+    @RolesAllowed("efiling-user")
+    public ResponseEntity<FilingPackage> getSubmissionFilingPackage(UUID xAuthUserId, UUID submissionId) {
+        Optional<Submission> fromCacheSubmission = this.submissionStore.get(submissionId, xAuthUserId);
```
Comment (karthikmsqa) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:230]: Confirmation question: 
'xAuthUserId' doesn't need to be changed to 'xTransactionId'?

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java [hunk 4]
```diff
@@ -240,12 +237,12 @@ private UserDetails buildUserDetails(Submission submission) {
 
     @Override
     @RolesAllowed("efiling-user")
-    public ResponseEntity<SubmitFilingPackageResponse> submit(UUID xAuthUserId, UUID id, SubmitFilingPackageRequest submitFilingPackageRequest) {
+    public ResponseEntity<SubmitFilingPackageResponse> submit(UUID xAuthUserId, UUID submissionId, SubmitFilingPackageRequest submitFilingPackageRequest) {
```
Comment (karthikmsqa) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:240]: Same as above: 
'xAuthUserId' doesn't need to be changed here to be 'xTransactionId'?

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java [hunk 5]
```diff
@@ -240,12 +237,12 @@ private UserDetails buildUserDetails(Submission submission) {
...
         ResponseEntity response;
-        MDC.put(Keys.EFILING_SUBMISSION_ID, id.toString());
+        MDC.put(Keys.EFILING_SUBMISSION_ID, submissionId.toString());
         //TODO: this will get the submission details from the cache and build a submission object
         try {
-            SubmitFilingPackageResponse result = submissionService.submitFilingPackage(xAuthUserId, id, submitFilingPackageRequest);
+            SubmitFilingPackageResponse result = submissionService.submitFilingPackage(xAuthUserId, submissionId, submitFilingPackageRequest);
```
Comment (karthikmsqa) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:245]: Same as above: 
'xAuthUserId' doesn't need to be changed here to be 'xTransactionId'?

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java (hunk 3 above)
Comment (alexjoybc) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:230]: yup yup nice catch

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java (hunk 4 above)
Comment (alexjoybc) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java]: same ;-)

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java [hunk 6]
```diff
@@ -109,15 +109,17 @@ public SubmissionApiDelegateImpl(
 
     @Override
     @RolesAllowed("efiling-user")
-    public ResponseEntity<Resource> getSubmissionDocument(UUID xAuthUserId, UUID id, String filename) {
+    public ResponseEntity<Resource> getSubmissionDocument(UUID xTransactionId,
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionApiDelegateImpl.java:112]: aha nice

**Pull request merged**